### PR TITLE
Yet another icds copy app command

### PR DESCRIPTION
**Pull request by orangejenny**
##### SUMMARY
https://dimagi-dev.atlassian.net/browse/ICDS-919

Reworking of https://github.com/dimagi/commcare-hq/pull/26482, which doesn't work because it attempted to do a remote request from a master app to its linked app, which isn't supported (`DomainLink` doesn't contain remote details for master apps).

This code is simpler and deletes much of the previous PR. The downside is that copying a linked app for ICDS now requires running multiple management commands instead of one.

Review by commit.

##### FEATURE FLAG
Mulltimaster linked apps + Multimaster linked apps bypass CC version check

##### PRODUCT DESCRIPTION
Don't announce.

### CODE REVIEW

File: corehq/apps/app_manager/forms.py [hunk 1]
```diff
@@ -69,6 +69,11 @@ def clean_domain(self):
         domain_obj = Domain.get_by_name(domain)
         if domain_obj is None:
             raise forms.ValidationError("A valid project space is required.")
+        if toggles.MULTI_MASTER_BYPASS_VERSION_CHECK.enabled(domain):
+            raise forms.ValidationError("""
+                Copying an app to a domain the uses multi-master linked apps and bypasses
```
Comment (mkangia) [inline corehq/apps/app_manager/forms.py]: that*?

File: corehq/apps/app_manager/forms.py [hunk 2]
```diff
@@ -69,6 +69,11 @@ def clean_domain(self):
...
+                the minimum CommCare version check requires developer intervention.
```
Comment (mkangia) [inline corehq/apps/app_manager/forms.py]: check which* requires?

File: custom/icds/management/commands/copy_icds_app_2_of_2.py [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+
+        After running copy_icds_app_1_of_2 on the master domain to create a copy of the master app,
+        run this command on all apps linked to that master app.
+
+        Looks up any ResourceOverrides for the given linked app that are tied to the original
+        master app's form ids. For any such override, the script creates a new ResourceOverride
+        where the pre_id is the unique id of that same form in the master app copy and the
+        post_id is the same as the original override.
+
+        This command is necessary until the ICDS apps are pinned to 2.47.4+.
```
Comment (mkangia) [inline custom/icds/management/commands/copy_icds_app_2_of_2.py]: just curious, what does pinned to 2.47.4+ imply?

**Review by mkangia — APPROVED** [review]

File: corehq/apps/app_manager/forms.py (hunk 2 above)
Comment (orangejenny) [reply corehq/apps/app_manager/forms.py]: that one's intentional

File: custom/icds/management/commands/copy_icds_app_2_of_2.py (hunk 3 above)
Comment (orangejenny) [reply custom/icds/management/commands/copy_icds_app_2_of_2.py]: It fixes the bug where forms crash on mobile if their unique id gets changed. If everyone were on 2.47.4+, we wouldn't need the `ResourceOverride` code at all.

Comment (mkangia) [reply custom/icds/management/commands/copy_icds_app_2_of_2.py]: okay thanks
so it's simply `until the ICDS users are on 2.47.4+.`?
When i read pinned, i thought we were restricting users to some version.

File: corehq/apps/app_manager/forms.py (hunk 2 above)
Comment (mkangia) [reply corehq/apps/app_manager/forms.py]: Hmm, i would see this as

```
Copying an app to a domain *that* uses multi-master linked apps *which* bypasses
the minimum CommCare version check*. This* requires developer intervention.
```

File: custom/icds/management/commands/copy_icds_app_2_of_2.py (hunk 3 above)
Comment (orangejenny) [reply custom/icds/management/commands/copy_icds_app_2_of_2.py]: It is indeed just `until the ICDS users are on 2.47.4+`, but the way we would know/enforce that all users are on 2.47.4+ is by restricting the version, right?

File: corehq/apps/app_manager/forms.py (hunk 2 above)
Comment (orangejenny) [reply corehq/apps/app_manager/forms.py]: I don't think that makes sense, let's offline if you think the existing wording doesn't make sense.

File: custom/icds/management/commands/copy_icds_app_2_of_2.py (hunk 3 above)
Comment (mkangia) [reply custom/icds/management/commands/copy_icds_app_2_of_2.py]: you mean all users are on commcare version `2.47.4` or more.
I don't think there is a way to be sure about that ever. 
What do you mean by `restricting the version`?

Comment (orangejenny) [reply custom/icds/management/commands/copy_icds_app_2_of_2.py]: By restricting, I mean only allow users to update to the next AWW version if they're on 2.47.4. I don't remember what mechanism we have for this - prompted updates, maybe? - but I'm certain we have one.

Comment (mkangia) [reply custom/icds/management/commands/copy_icds_app_2_of_2.py]: i guess something related to setting the app version in the app.
and even in that, not clear what the message phone would complain about and if user would actually make an effort to update.
If you are clear about it, fine to abandon this conversation.

Comment (orangejenny) [reply custom/icds/management/commands/copy_icds_app_2_of_2.py]: Oh, you're right, it's just a matter of setting the CC version in the app. I think we're okay to drop this conversation.

**Pull request merged**